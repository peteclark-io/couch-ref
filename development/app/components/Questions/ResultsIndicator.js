'use strict';

import React from 'react';
import {connect} from 'react-redux';

import SimpleLine from '../Charts/SimpleLine';
import styles from './styles.css';

const ResultsIndicator = React.createClass({

   propTypes: {
     data: React.PropTypes.arrayOf(React.PropTypes.object)
   },

   render: function() {
       if (!this.props.data){
         return null;
       }

      return (
         <div className={styles['results-indicator']}>
            <SimpleLine data={this.props.data} />
         </div>
      );
   }
});

const getQuestionResults = (state = {statistics: {}}, id) => {
  if (state.statistics[id]){
    return [{title: 'Yes', value: state.statistics[id].simple.yes}, {title: 'No', value: state.statistics[id].simple.no}];
  }
  return undefined;
};

const mapStateToProps = (state, ownProps) => {
   return {
     data: getQuestionResults(state, ownProps.id)
   };
};

const LiveResultsIndicator = connect(
  mapStateToProps
)(ResultsIndicator);

export default LiveResultsIndicator;
