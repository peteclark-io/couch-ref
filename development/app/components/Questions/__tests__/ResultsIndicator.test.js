'use strict';

import React from 'react';
import ResultsIndicator from '../ResultsIndicator';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

it('Is null when it has no data.', () => {
   const rendered = mount(
      <ResultsIndicator />
   );
   expect(rendered.html()).toBeNull();
});

it('Writes a graph when it has data.', () => {
   const data = {simple: {yes: 1, no: 1}};
   const rendered = mount(
      <ResultsIndicator statistic={data} />
   );
   expect(rendered.html()).toMatchSnapshot();
});

it('Is null when data is 0.', () => {
   const data = {simple: {yes: 0, no: 0}};
   const rendered = mount(
      <ResultsIndicator statistic={data} />
   );
   expect(rendered.html()).toMatchSnapshot();
});
