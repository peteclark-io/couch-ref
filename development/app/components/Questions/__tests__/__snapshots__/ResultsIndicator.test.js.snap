exports[`test Is null when data is 0. 1`] = `"<div class=\"results-indicator\"><!-- react-empty: 2 --></div>"`;

exports[`test Writes a graph when it has data. 1`] = `"<div class=\"results-indicator\"><svg width=\"100%\" height=\"24px\"><g transform=\"translate(0,8)\"><text dy=\"-1\" style=\"font-size: 10px;\">Yes</text><rect height=\"3\" fill=\"#66d3e4\" width=\"NaN\"></rect></g><g transform=\"translate(0,20)\"><text dy=\"-1\" style=\"font-size: 10px;\">No</text><rect height=\"3\" fill=\"rgba(60,90,150,1)\" width=\"NaN\"></rect></g></svg></div>"`;
